### PR TITLE
docs: extract DateFacet documentation

### PR DESCRIPTION
**Pull request by samisayegh**
https://coveord.atlassian.net/browse/KIT-503

### CODE REVIEW

File: packages/headless/src/controllers/facets/range-facet/date-facet/date-range.ts [hunk 1]
```diff
@@ -2,29 +2,59 @@ import dayjs from 'dayjs';
 import utc from 'dayjs/plugin/utc';
 import customParseFormat from 'dayjs/plugin/customParseFormat';
 import {DateRangeRequest} from '../../../../features/facets/range-facets/date-facet-set/interfaces/request';
+import {FacetValueState} from '../../../../features/facets/facet-api/value';
 
 dayjs.extend(utc);
 dayjs.extend(customParseFormat);
 
-type DateType = string | number | Date;
+export type DateValueType = string | number | Date;
```
Comment (samisayegh) [inline packages/headless/src/controllers/facets/range-facet/date-facet/date-range.ts:10]: Any suggestions around the name of this type alias? I felt `DateType` was too general, but `DateRangeValueType` was too confusing.

Comment (olamothe) [reply packages/headless/src/controllers/facets/range-facet/date-facet/date-range.ts:10]: I'm not particularly fond on the word `Type` in something that is... a type.

Suggestions: `DateInput`, `DateConfig`.

As in, you use this to "input" or "configure" a date.

**Review by olamothe — APPROVED** [review]

**Review by ThibodeauJF — APPROVED** [review]

File: packages/headless/src/features/facets/range-facets/date-facet-set/interfaces/request.ts [hunk 2]
```diff
@@ -1,10 +1,31 @@
...
+export interface DateRangeRequest {
+  /**
+   * The start value of the range.
+   */
+  start: string;
+
+  /**
+   * The end value of the range.
+   */
+  end: string;
```
Comment (btaillon) [inline packages/headless/src/features/facets/range-facets/date-facet-set/interfaces/request.ts:17]: We should probably include the date format in those comments.

File: packages/headless/src/features/facets/range-facets/date-facet-set/interfaces/response.ts [hunk 3]
```diff
@@ -1,5 +1,31 @@
...
+
+  /**
+   * The starting value for the date range.
+   */
+  start: string;
+
+  /**
+   * The ending value for the date range.
+   */
+  end: string;
```
Comment (btaillon) [inline packages/headless/src/features/facets/range-facets/date-facet-set/interfaces/response.ts:18]: Same as above

**Review by btaillon — APPROVED** [review]

File: packages/headless/src/features/facets/range-facets/date-facet-set/interfaces/request.ts (hunk 2 above)
Comment (samisayegh) [reply packages/headless/src/features/facets/range-facets/date-facet-set/interfaces/request.ts:17]: Great suggestion, thanks 💯

**Pull request merged**